### PR TITLE
add fixture for texting msgpack decoding of binary protocol

### PR DESCRIPTION
**Pull request by amnonbc**

### CODE REVIEW

File: test-resources/msgpack_test_fixtures.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
+[
+  {
+    "Name": "json array",
```
Comment (lmars) [inline test-resources/msgpack_test_fixtures.json:3]: The other JSON files in this directory use camelCase for the keys, should this file conform to that convention?

File: test-resources/msgpack_test_fixtures.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
+[
```
Comment (lmars) [inline test-resources/msgpack_test_fixtures.json:1]: I think we should document in the feature spec (specifically in [RSL6a1](https://sdk.ably.com/builds/ably/specification/main/features/#RSL6a1)) how we expect client libraries to use these fixtures (and perhaps put a comment in this file linking to it).

File: test-resources/msgpack_test_fixtures.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  {
+    "Name": "json array",
+    "Data": ["a", "b", "c"],
+    "Encoding": "json",
+    "NumRepeat": 0,
+    "Type": "jsonArray",
+    "MsgPack": "g7Bjb25uZWN0aW9uU2VyaWFsAKhtZXNzYWdlc5GCpGRhdGGtWyJhIiwiYiIsImMiXahlbmNvZGluZ6lqc29uL2pzb26pbXNnU2VyaWFsAA=="
```
Comment (lmars) [inline test-resources/msgpack_test_fixtures.json:8]: We should document that this is the msgpack encoding of a ProtocolMessage, not just a Message

Comment (lmars) [reply test-resources/msgpack_test_fixtures.json:8]: Also, when I msgpack decode this, I get:

```json
{
  "connectionSerial": 0,
  "messages": [
    {
      "data": "[\"a\",\"b\",\"c\"]",
      "encoding": "json/json"
    }
  ],
  "msgSerial": 0
}
```

Should the encoding be `json`, not `json/json`?

File: test-resources/msgpack_test_fixtures.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  {
+    "Name": "short string",
+    "Data": "a",
+    "Encoding": "",
+    "NumRepeat": 1,
+    "Type": "string",
+    "MsgPack": "g7Bjb25uZWN0aW9uU2VyaWFsAKhtZXNzYWdlc5GBpGRhdGGhYaltc2dTZXJpYWwA"
+  },
+  {
+    "Name": "200 byte string string",
```
Comment (lmars) [inline test-resources/msgpack_test_fixtures.json:19]:
```suggestion
    "Name": "200 character string",
```

File: test-resources/msgpack_test_fixtures.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    "Data": "a",
+    "Encoding": "",
+    "NumRepeat": 200,
+    "Type": "string",
+    "MsgPack": "g7Bjb25uZWN0aW9uU2VyaWFsAKhtZXNzYWdlc5GBpGRhdGHZyGFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhqW1zZ1NlcmlhbAA="
+  },
+  {
+    "Name": "2000 byte string string",
```
Comment (lmars) [inline test-resources/msgpack_test_fixtures.json:27]:
```suggestion
    "Name": "2000 character string",
```

File: test-resources/msgpack_test_fixtures.json (hunk 1 above)
Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json:3]: you mean snake_case

Yes, fair point. I'll fix it

Comment (lmars) [reply test-resources/msgpack_test_fixtures.json:3]: I did mean camelCase (e.g. [`expectedType`](https://github.com/ably/ably-common/blob/main/test-resources/messages-encoding.json#L7) or [`pushEnabled`](https://github.com/ably/ably-common/blob/main/test-resources/test-app-setup.json#L22), as opposed to PascalCase `ExpectedType` or `PushEnabled`)

Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json]: done

File: test-resources/msgpack_test_fixtures.json (hunk 2 above)
Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json:1]: OK. But that should be in a separate PR.

File: test-resources/msgpack_test_fixtures.json (hunk 3 above)
Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json]: > We should document that this is the msgpack encoding of a ProtocolMessage, not just a Message

But where should that be documented? json does not alllow comments.

Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json]: > Also, when I msgpack decode this, I get:
> 
> ```json
> {
>   "connectionSerial": 0,
>   "messages": [
>     {
>       "data": "[\"a\",\"b\",\"c\"]",
>       "encoding": "json/json"
>     }
>   ],
>   "msgSerial": 0
> }
> ```
> 
> Should the encoding be `json`, not `json/json`?

well caught...
fixed in [da9c383](https://github.com/ably/ably-common/pull/181/commits/da9c383c56b3eafb14e9d5339199d2ccb238e9f9)

Comment (lmars) [reply test-resources/msgpack_test_fixtures.json]: > But where should that be documented? json does not alllow comments.

In the other fixture files, we put a comment in a key starting with an underscore (e.g. [here](https://github.com/ably/ably-common/blob/main/test-resources/messages-encoding.json#L2) and [here](https://github.com/ably/ably-common/blob/main/test-resources/test-app-setup.json#L2)), and we could put it in the [README](https://github.com/ably/ably-common/tree/main/test-resources#readme), and we could put it in the [spec](https://sdk.ably.com/builds/ably/specification/main/features/#RSL6a1).

File: test-resources/msgpack_test_fixtures.json (hunk 2 above)
Comment (lmars) [reply test-resources/msgpack_test_fixtures.json:1]: Yes the feature spec is a separate repo so would need a separate PR, but I think it's worth doing as part of the work your doing so other SDK developers know these fixtures exist and how to test against them.

File: test-resources/msgpack_test_fixtures.json (hunk 3 above)
Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json]: added to the README.md in 5725a60

File: test-resources/msgpack_test_fixtures.json (hunk 4 above)
Comment (amnonbc) [reply test-resources/msgpack_test_fixtures.json]: done

**Review by lmars — APPROVED** [review]

**Pull request merged**